core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,286 @@
+      }
+      if (representation != null) {
+        if (titleBlock instanceof DAnnotation) {
+          int numLines = ((DAnnotation) titleBlock).getReferences().size();
+          List<EObject> lines = ((DAnnotation) titleBlock).getReferences();
+          for (int i = 0; i < numLines; i++) {
+            DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+            annotationCol.setSource("TitleBlockLineCol");
+            if (((DAnnotation) titleBlock).getReferences().get(i) instanceof DAnnotation) {
+              ((DAnnotation) (((DAnnotation) titleBlock).getReferences().get(i))).getReferences().add(annotationCol);
+              representation.getEAnnotations().add(annotationCol);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private boolean isDiagramLevel(EObject titleBlock) {
+    for (EObject reference : ((DAnnotation) titleBlock).getReferences()) {
+      if (!(reference instanceof DAnnotation)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private void addColumnsToLine(DAnnotation annotationLine, DRepresentation representation, int numCols,
+      int lineNumber) {
+    List<DAnnotation> annotationCols = new ArrayList<DAnnotation>();
+    for (int j = 0; j < numCols; j++) {
+      DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+      annotationCol.setSource("TB_" + lineNumber + "_" + j);
+      annotationCols.add(annotationCol);
+      representation.getEAnnotations().add(annotationCol);
+    }
+    annotationLine.getReferences().addAll(annotationCols);
+    representation.getEAnnotations().add(annotationLine);
+  }
+
+  public int getNumLines(DAnnotation titleBlock) {
+    return titleBlock.getReferences().size();
+  }
+
+  public int getNumColumns(DAnnotation titleBlock) {
+    EObject obj = titleBlock.getReferences().get(1);
+    if (obj instanceof DAnnotation)
+      return ((DAnnotation) obj).getReferences().size();
+    return 0;
+  }
+
+  public Collection<EObject> computeTitleBlockElements(EObject elem, EObject diagram) {
+    Collection<EObject> result = new ArrayList<>();
+
+    if ((diagram instanceof DRepresentation)) {
+      DRepresentation representation = (DRepresentation) diagram;
+      result = representation.getEAnnotations().stream().filter(x -> x.getReferences().contains(elem))
+          .collect(Collectors.toList());
+    }
+
+    return result;
+  }
+
+  public List<DAnnotation> checkIsTitleBlockContainer(Object elementView) {
+    List<DAnnotation> list = new ArrayList<DAnnotation>();
+    if ((elementView instanceof DRepresentation)) {
+      DRepresentation representation = (DRepresentation) elementView;
+      list = representation.getEAnnotations().stream().filter(x -> x.getSource().equals("TitleBlock"))
+          .collect(Collectors.toList());
+    }
+    return list;
+  }
+
+  public List<Object> getTitleBlockCellContent(EObject diagram, EObject cell) {

```

Comment:
As I observe, this is evaluated on the graphical elements but not the semantic elements. For example, typing feature:summary on a cell of a title block linked to a class does not show anything though a class does have a summary.

Reply:
![summary](https://user-images.githubusercontent.com/60213999/73925765-d27b9a00-48ce-11ea-8b91-f072aee6641b.png)
It appears that is actually working.

Reply:
There's a little bug when the element does not have the feature (e.g. summary), the cell shows some weird elements...

Reply:
Fixed.